### DISCUSSION TITLE
"Status" field not reflecting offline duration properly

### DISCUSSION THREAD

**Discussion by n9yty** [sinamics/ztnet#311] [General]
I have a group of computers on a zerotier network managed by a zerotier controller and they are also joined to a ztnet controller. This makes it easy to test things out before transitioning away from ZeroTier in case they charge the prices I've read about on reddit that some are getting quoted.

On the ztnet controller, though, I notice the Status column will show "offline 1 week" (various durations) and yet I go over to the ZeroTier managed controller and the time is reported considerably different, typically offline a day or less, usually just a few hours. This would be in line with how the computers are likely being used.

Why doesn't ztnet properly update/reflect the last connected time? Does it depend on the web page being reloaded/refreshed to capture that information?

**Comment by tinola** [comment]
I think it's not possible without some cron job running in the background for example every 5 minutes, that would update the status of members in the database. When you have the ztnet application open, the "last seen" status is updated continuously in the database.

**Comment by sinamics** [comment]
Peer information is fetched/updated when you open the network page in ztnet. To ensure the accuracy of this information, we would need to add a background task like @tinola is mentioning.

**Comment by n9yty** [comment]
I would be open to doing that, is this something you might plan to handle, or should I look at implementing it?  I'm not sure what it would have to do or what it would look like, but I would like to have the information available in the same way it is on ZeroTier's controller.

**Comment by sinamics** [comment]
You are free to give it a go. It can be added to the cronTasks.ts
https://github.com/sinamics/ztnet/blob/main/src/cronTasks.ts

Then call the cron job in `instrumentation.ts`
https://github.com/sinamics/ztnet/blob/main/src/instrumentation.ts


If not, I can add it when I have time, in a week or so. 
I'm currently working on another project that will soon be completed.

**Comment by n9yty** [comment]
Wow, thanks for the information, but I think this is outside my scope right now, I'm not familiar enough with the platform. I was thinking of implementing it if it could be constructed in an external cron job, but I can't work out any way to have it authenticate to the right page to keep it refreshed, which also would be very inefficient. :)

It would be great if this could be implemented whenever you are working on the project. It is not a functional problem so there is no urgency, although it would be nice to have it.

**Comment by n9yty** [comment]
By the way, this seems to be working great! Thanks for handling it so quickly!!  I am on a github clone/self-built release, so if this isn't released yet it is only a matter of time. :)
